src/bin/build_octree.rs

```diff
@@ -123,7 +142,7 @@ fn split_node<'a, 'b: 'a, P>(
     let (leaf_nodes, split_nodes): (Vec<_>, Vec<_>) =
         children
             .into_iter()
-            .partition(|n| n.num_points < MAX_POINTS_PER_NODE);
+            .partition(|n| !should_split_node(n, resolution));
```

Comment:
I had this originally, but found that the function is harder to think about then. So I opted for the inversion here.